src/vignette/util/image_response.clj

```diff
@@ -38,32 +43,39 @@
 
 (defn create-image-response
   ([image image-map]
-   (-> (response (->response-object image))
-       (header "Content-Type" (content-type image))
-       (header "Content-Length" (content-length image))
-       (header "ETag" (etag image))
-       (header "X-Thumbnailer" "Vignette")
-       (add-content-disposition-header image-map)
-       (add-surrogate-header image-map)))
+   (as-> (response (->response-object image)) resp
+         (when-header-val resp "Content-Type" (content-type image))
+         (when-header-val resp "Content-Length" (content-length image))
+         (when-header-val resp "ETag" (etag image))
+         (header resp "X-Thumbnailer" "Vignette")
+         (add-content-disposition-header resp image-map image)
+         (add-surrogate-header resp image-map)))
   ([image]
     (create-image-response image nil)))
 
 (defn create-head-response
   [image-map]
   (-> (response "")
-       ;(header "Content-Type" (content-type image))
        (header "X-Thumbnailer" "Vignette")
        (add-content-disposition-header image-map)
        (add-surrogate-header image-map)))
 
+(defn- base-filename [image-map object]
+  (or (if-let [orig (if image-map (original image-map))]
+        (string/replace orig "\"" "\\\""))
+      (when object (filename object))))
+
 (defn add-content-disposition-header
-  [response-map image-map]
-  (if (original image-map)
-    (let [requested-format (query-opt image-map :format)
-          filename (cond-> (string/replace (original image-map) "\"" "\\\"")
-                     requested-format (str "." requested-format))]
-      (header response-map "Content-Disposition" (format "inline; filename=\"%s\"" filename)))
-    response-map))
+  ([response-map image-map object]
```

Comment:
What is the object parameter? It's not intuitive to me why the additional object is required.